src/clojure/nrepl/socket.clj

```diff
@@ -0,0 +1,222 @@
+(ns nrepl.socket
+  "Compatibility layer for java.io vs java.nio sockets to allow an
+  incremental transition to nio, since the JDK's filesystem sockets
+  don't support the java.io socket interface, and we can't use the
+  compatibility layer for bidirectional read and write:
+  https://bugs.openjdk.java.net/browse/JDK-4509080."
+  (:require
+   [clojure.java.io :as io]
+   [nrepl.misc :refer [log]])
+  (:import
+   (clojure.lang Reflector)
+   (java.io BufferedInputStream BufferedOutputStream File)
+   (java.net InetSocketAddress ServerSocket Socket StandardProtocolFamily URI)
+   (java.nio ByteBuffer)
+   (java.nio.file Path)
+   (java.nio.channels Channels ClosedChannelException
+                      ServerSocketChannel SocketChannel)))
+
+(def orig-warn-on-reflection *warn-on-reflection*)
+
+(defmacro find-class [full-path]
+  `(try
+     (import ~full-path)
```

Comment:
I think `Class/forName` would be a better / more standard choice here.

Reply:
Thanks, fix pending.